src/ports/postgres/modules/convex/mlp_igd.py_in

```diff
@@ -0,0 +1,754 @@
+                                          array_type=dependent_type)
+        # TODO validation for summary
+        summary_table_creation_query = """
+            CREATE TABLE {output_table}_summary(
+                source_table TEXT,
+                independent_varname TEXT,
+                dependent_varname TEXT,
+                tolerance FLOAT,
+                step_size FLOAT,
+                n_iterations INTEGER,
+                n_tries INTEGER,
+                layer_sizes INTEGER[],
+                activation_function TEXT,
+                is_classification BOOLEAN,
+                classes {dependent_type}[]
+            )""".format(output_table=output_table,
+                        dependent_type=dependent_type)
+
+        summary_table_update_query = """
+            INSERT INTO {output_table}_summary VALUES(
+                '{source_table}',
+                '{independent_varname}',
+                '{original_dependent_varname}',
+                {tolerance},
+                {step_size},
+                {n_iterations},
+                {n_tries},
+                {layer_sizes_str},
+                '{activation_name}',
+                {is_classification},
+                {classes_str}
+            )
+            """.format(**locals())
+        plpy.execute(summary_table_creation_query)
+        plpy.execute(summary_table_update_query)
+# ----------------------------------------------------------------------
+
+
+def _build_model_table(schema_madlib, output_table, final_state, n_iterations):
+    final_state_str = py_list_to_sql_string(final_state, array_type="double precision")
+
+    model_table_query = """
+        CREATE TABLE {output_table} AS
+            SELECT
+                (result).coeff AS coeff,
+                (result).loss  AS loss,
+                {n_iterations} AS num_iterations
+                -- (result).num_rows_processed     AS num_rows_processed,
+                -- n_tuples_including_nulls - (result).num_rows_processed
+            FROM (
+                SELECT
+                    {schema_madlib}.internal_mlp_igd_result(
+                        {final_state_str}
+                    ) AS result
+            ) rel_state_subq
+        """.format(**locals())
+    plpy.execute(model_table_query)
+# ----------------------------------------------------------------------
+
+
+def _get_optimizer_params(param_str):
+    params_defaults = {
+        "step_size": (0.001, float),
+        "n_iterations": (100, int),
+        "n_tries": (1, int),
+        "tolerance": (0.001, float),
+    }
+    param_defaults = dict([(k, v[0]) for k, v in params_defaults.items()])
+    param_types = dict([(k, v[1]) for k, v in params_defaults.items()])
+
+    if not param_str:
+        return param_defaults
+
+    name_value = extract_keyvalue_params(param_str, param_types, param_defaults,
+                                         ignore_invalid=True)
```

Comment:
Is there a specific reason why invalid optimizer parameters are ignored (ignore_invalid=True)?